pkg/controllers/resources/namespaces/syncer.go

```diff
@@ -96,7 +104,28 @@ func (s *namespaceSyncer) Sync(ctx *synccontext.SyncContext, event *synccontext.
 
 func (s *namespaceSyncer) SyncToVirtual(ctx *synccontext.SyncContext, event *synccontext.SyncToVirtualEvent[*corev1.Namespace]) (_ ctrl.Result, retErr error) {
 	// virtual object is not here anymore, so we delete
-	return patcher.DeleteHostObject(ctx, event.Host, nil, "virtual object was deleted")
+	if event.VirtualOld != nil || translate.ShouldDeleteHostObject(event.Host) {
```

Comment:
Same as above, seems like you should enable objectcaching

Reply:
enabled 👍